pysb/generator/bng.py

```diff
@@ -249,14 +251,31 @@ def warn_caller(message):
         module = inspect.getmodule(caller_frame)
     warnings.warn(message, stacklevel=stacklevel)
 
+
+class BngPrinter(StrPrinter):
+    def __init__(self, **settings):
+        super(BngPrinter, self).__init__(settings)
+
+    def _print_Piecewise(self, expr):
+        if type(expr.args[-1][1]) != BooleanTrue:
```

Comment:
I thought so too, however it turns out `BooleanTrue` has type `sympy.core.singleton.Singleton`, so the proposed comparison fails.